backend/packages/Upgrade/src/api/middlewares/UserCheckMiddleware.ts

```diff
@@ -12,8 +12,16 @@ export class UserCheckMiddleware {
   public async use(req: AppRequest, res: AppRequest, next: express.NextFunction): Promise<any> {
     try {
       const user_id = req.get('User-Id');
-      req.logger.child({ user_id });
-      req.logger.debug({ message: 'User Id is:', user_id });
+      if (!user_id) {
+        const error = new Error(`User-Id header not found.`);
+        (error as any).type = SERVER_ERROR.USER_NOT_FOUND;
```

Comment:
well it's a bad request type, not really a user not found, the server has incomplete info to even try to find a user

Reply:
@danoswaltCL Yeah, I was also thinking to add a new ERROR type: `MISSING_HEADER_USER_ID`. But, got an input from our playpower team to reuse existing one's. I know this is for logged in user data not found from DB. I will update this if the above type looks good to you.

Reply:
<img width="851" alt="Screenshot 2024-09-26 at 2 04 58 PM" src="https://github.com/user-attachments/assets/abea127d-6b19-4754-b07f-2397ebab89f5">